src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift

```diff
@@ -376,6 +380,26 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, Re
 		)
 	}()
 
+	/// Prepare handling of scanned QR codes
+	///
+	/// - Parameter launchOptions: Launch options passed on app launch
+	/// - Returns: A `Route` if a valid URL is passed in the launch options
+	private func routeForScannedQRCode(in launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Route? {
+		guard let activityDictionary = launchOptions?[.userActivityDictionary] as? [AnyHashable: Any] else {
+			return nil
+		}
+
+		for key in activityDictionary.keys {
+			if let userActivity = activityDictionary[key] as? NSUserActivity,
+			   userActivity.activityType == NSUserActivityTypeBrowsingWeb,
+			   let url = userActivity.webpageURL {
+				return Route(url: url)
+			}
+		}
+
+		return nil
```

Comment:
why not move it to the Router struct? Somthing like init?(with launchOptions: ...)

Reply:
It would reduce clutter in the AppDelegate but I think it's not the right place. We already have two initializers in `Route` and would create a third one. Also we have to 'soften' up the scope of `Route` by importing UIKit and tailor this initializer to AppDelegates while ignoring the different handling in scenes (yes, I know, we'll never see them again in this project).

Long story short: as much as I would reduce the amount of code in our app delegate, I don't think this is a good part to move out (expect another extension)